tests/test_model_saving.py

```diff
@@ -102,6 +103,34 @@ def test_functional_model_saving():
 
 
 @keras_test
+def test_model_saving_to_file_descriptor():
+    input = Input(shape=(3,))
+    x = Dense(2)(input)
+    output = Dense(3)(x)
+
+    model = Model(input, output)
+    model.compile(loss=losses.MSE,
+                  optimizer=optimizers.RMSprop(lr=0.0001),
+                  metrics=[metrics.categorical_accuracy])
+    x = np.random.random((1, 3))
+    y = np.random.random((1, 3))
+    model.train_on_batch(x, y)
+
+    out = model.predict(x)
+    _, fname = tempfile.mkstemp('.h5')
+    with h5py.File(fname, 'w') as h5file:
+        group = h5file.create_group('model')
```

Comment:
Can you save something else in the file as well? Random garbage, and maybe another model, to make sure we are loading the correct one.